pkgs/by-name/ol/olympus/package.nix

```diff
@@ -0,0 +1,172 @@
+  # - writeShellScriptBin { ... }: Use a custom script.
+  # - ./my-wrapper.sh: Use a custom script.
+  # In any case, it can be overridden at runtime by OLYMPUS_{CELESTE,LOENN,MINIINSTALLER}_WRAPPER.
+  celesteWrapper ? null,
+  loennWrapper ? null,
+  miniinstallerWrapper ? null,
+  skipHandlerCheck ? false, # whether to skip olympus xdg-mime check, true will override it
+}:
+
+let
+  lua_cpath =
+    with luajitPackages;
+    lib.concatMapStringsSep ";" getLuaCPath [
+      (buildLuarocksPackage {
+        pname = "lsqlite3";
+        version = "0.9.6-1";
+        src = fetchzip {
+          url = "http://lua.sqlite.org/index.cgi/zip/lsqlite3_v096.zip";
+          hash = "sha256-Mq409A3X9/OS7IPI/KlULR6ZihqnYKk/mS/W/2yrGBg=";
+        };
+        buildInputs = [ sqlite.dev ];
+      })
+
+      lua-subprocess
+      nfd
+    ];
+
+  # When installing Everest, Olympus uses MiniInstaller, which is dynamically linked.
+  miniinstaller-fhs = buildFHSEnv {
+    pname = "olympus-miniinstaller-fhs";
+    meta.mainProgram = "${miniinstaller-fhs.pname}"; # so lib.getExe works
+    inherit version;
+    targetPkgs =
+      pkgs:
+      (with pkgs; [
+        icu
+        openssl
+        dotnet-runtime # Without this, MiniInstaller will install dotnet itself.
+      ]);
+  };
+
+  wrapper-to-env =
+    wrapper:
+    if lib.isDerivation wrapper then
+      lib.getExe wrapper
+    else if wrapper != null then
+      wrapper
+    else
+      "";
+
+  miniinstaller-wrapper =
+    if miniinstallerWrapper == null then
+      (writeShellScript "miniinstaller-wrapper" "exec ${lib.getExe miniinstaller-fhs} -c \"$@\"")
+    else
+      (wrapper-to-env miniinstallerWrapper);
+
+  pname = "olympus";
+  phome = "$out/lib/${pname}";
+  # The following variables are to be updated by the update script.
+  version = "24.11.23.03";
+  buildId = "4420"; # IMPORTANT: This line is matched with regex in update.sh.
+  rev = "a3792e0c85f3ad7a3029a6a66ca8288aa6f58ae4";
+
+in
+buildDotnetModule {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "EverestAPI";
+    repo = "Olympus";
+    fetchSubmodules = true; # Required. See upstream's README.
+    hash = "sha256-UPAn9Rbm2IlxMJ/O69WXHugIc+22w+B5i6iLkCcsfQ8=";
+  };
+
+  nativeBuildInputs = [
+    libarchive # To create the .love file (zip format).
+  ];
+
+  nugetDeps = ./deps.nix;
+  projectFile = "sharp/Olympus.Sharp.csproj";
+  executables = [ ];
+
+  # See the 'Dist: Update src/version.txt' step in azure-pipelines.yml from upstream.
+  preConfigure = ''
+    echo ${version}-nixos-${buildId}-${builtins.substring 0 5 rev} > src/version.txt
+  '';
+
+  # Hack Olympus.Sharp.bin.{x86,x86_64} to use system mono.
+  # This was proposed by @0x0ade on discord.gg/celeste.
+  # https://discord.com/channels/403698615446536203/514006912115802113/827507533962149900
+  postBuild = ''
+    dotnet_out=sharp/bin/Release/net452
+    dotnet_out=$dotnet_out/$(ls $dotnet_out)
```

Comment:
I recently learned that we should avoid parsing the output of `ls` ([source](https://mywiki.wooledge.org/ParsingLs)).